pytorch_lightning/strategies/ddp.py

```diff
@@ -152,6 +154,17 @@ def setup(self, trainer: "pl.Trainer") -> None:
         if trainer_fn == TrainerFn.FITTING:
             self.configure_ddp()
 
+        # set up optimizers after the wrapped module has been moved to the device
+        self.setup_optimizers(trainer)
```

Comment:
Let's try to make the code more readable.

Let's call super().setup(trainer) and override `setup_optimizers` to add the post_localSGD within its own method.

```
super().setup(trainer)

def setup_optimizers(self):
    super().setup_optimizers()
    self._reinit_optimizers_with_post_localSGD() # contain the torch version checks
```